Add new Spring Boot tutorial (#5523)
JetBrains/kotlin-web-site: docs/topics/ai/spring-boot-claude.md

[//]: # (title: Create a task manager app with Spring Boot and Claude)

<web-summary>Learn how to create a Kotlin app with Claude and Spring Boot.</web-summary>

In this tutorial, you'll learn how to use [Claude](https://claude.com/product/overview) to create a Kotlin app to manage tasks. The tutorial uses Spring Boot

> to manage the backend infrastructure, while Claude handles the rest. 

Reviewer: (suggestion)
It sounds good as is - a bit marketingy which might not be an issue but WDYT about more specific like:  while Claude helps you plan, implement, test, and refine the application.

Author: I worry that "Claude helps you" makes it sound like you are still expected to do the work but really Claude is doing the work here. Would "Claude plans and develops the application."?

Reviewer: sounds good to me 👍

> > As with any AI-powered tool, Claude can make mistakes. Review Claude's changes and only use it with code you trust.

Reviewer: ```suggestion
> As with any AI-powered tool, Claude can make mistakes. Review Claude's changes carefully and apply only code you understand and trust.
```

Author: I half accepted this one. The second part is more about third party code and tools, not the code generated by Claude itself.

> For more information on Claude's security policy, see [Claude Code Docs](https://code.claude.com/docs/en/security).
> 
{style="note"}

## Set up the environment

> > This tutorial uses Claude through the JetBrains AI assistant, but you can also work through the tutorial steps using

Reviewer: ```suggestion
> This tutorial uses Claude through the JetBrains AI Assistant, but you can also work through the tutorial steps using
```

   * **Language**: Kotlin
   * **Type**: Gradle - Kotlin

     > This option specifies the build system and the DSL.
     >
     {style="tip"}

   * **Package name**: org.jetbrains.kotlin.taskmanagerdemo
   * **JDK**: jbr-21
   * **Java**: 17

     > If you don't have these Java and JDK versions installed, you can download them from the dropdown lists.
     >
     {style="tip"}

   ![Create Spring Boot project](create-spring-claude-project.png){width=800}

4. Make sure that you specified all the fields, then click **Next**.
5. Select the latest stable Spring Boot version in the **Spring Boot** field.
6. Select the **Web | Spring Web** dependency.

   ![Set up Spring Boot project](spring-claude-dependency.png){width=800}

7. Click **Create** to generate and set up the project.

   The IDE generates and opens the new project. It may take some time to download and import the project dependencies.

## Create a development plan

In your project:

1. Open the ![AI Chat](toolWindowChat@20x20.svg){width=20} **AI Chat** tool window. By default, the **Chat** mode is selected. Select **Claude Agent**.

   ![Select Claude Agent](select-claude-agent.png){width=300}

> 2. Click **Mode: Default**![Operation mode](app-client.expui.general.chevronDownLarge.svg){width=20}{type="joined"} and select **Mode: Plan Mode**.

Reviewer: ```suggestion
2. Click **Mode: Default** ![Operation mode](app-client.expui.general.chevronDownLarge.svg){width=20}{type="joined"} and select **Mode: Plan Mode**.
```

   Claude Agent is now ready to plan without executing actions.

   ![Select Plan Mode](claude-plan-mode.png){width=400}

   > For more information about the different operation modes, see [Select operation mode](https://www.jetbrains.com/help/ai-assistant/claude-agent.html#select-operation-mode).
   >
   {style="tip"}

3. Write a prompt asking Claude to create a task manager app. Share some details on what you think it should include. For example:

   ```text
   I'd like to create a task manager application for managing tasks, such as a grocery list. 
   It should have a basic UI and include categories, due dates, priorities, and status tracking. 

   Use VCS while working. Work step by step and create commits at each stage so I can review the changes afterward.
   ```

   > For guidance on how to design your prompt, see [Claude Code best practices](https://code.claude.com/docs/en/best-practices).
   >
   {style="tip"}

   Claude explores the existing project structure and suggests a plan.

4. Review the plan carefully before proceeding. If you want to make some modifications, select **No, keep planning** and share your follow-up comments.

> 5. When you're ready to proceed, select the **Yes ...** option that you are most comfortable with.

Reviewer: ```suggestion
5. When you're ready to proceed, select the **Yes ...** option that best matches how much control you want over Claude's changes.
```

Reviewer: (suggestion)
feels a bit more direct.

   ![Ready to code](ready-to-code.png){width=600}

   > For more information on the different options, see [Claude Code permission modes](https://code.claude.com/docs/en/best-practices).
   >
   {style="tip"}

> 6. Claude exits Plan Mode and begins working. Wait until it's complete.

Reviewer: ```suggestion
6. Claude exits Plan Mode and begins working. Wait until the task is complete.
```

Reviewer: not sure if we should make Plan Mode bold as well 🤔 WDYT?

## Review the commits

Before running the app, review the generated changes carefully:

> 1. Open the **Git** window to see the list of commits. 

Reviewer: ```suggestion
1. Open the **Git** tool window to see the list of commits. 
```

Reviewer: (suggestion) to align with how [intellij docs call it](https://www.jetbrains.com/help/idea/working-with-git-tutorial.html#create-test-prj)

2. Select a commit and double-click each modified file to review the diff in IntelliJ IDEA's side-by-side viewer.

> ![Side by side viewer](side-by-side-viewer.png){width=800}

Reviewer: ```suggestion
![Side-by-side viewer](side-by-side-viewer.png){width=800}
```

## Run the app

Once you are happy with the changes, run the app:

1. Run the `bootRun` Gradle task or enter the following command in the terminal:

   ```bash
   ./gradlew bootRun
   ```

> 2. In a browser, open the localhost URL. Typically, it's:

Reviewer: ```suggestion
2. In a browser, open the localhost URL. The default is usually:
```

   ```text
   http://localhost:8080
   ```

   You should now see the basic UI that Claude created.

   ![Run the app](run-spring-claude-app.png){width=800}

>    > Since Claude designs the UI, your UI may look different to the version in this tutorial.

Reviewer: ```suggestion
   > Since Claude designs the UI, your UI may look different from the version in this tutorial.
```

Reviewer: (suggestion) I think both are fine, but I think this is safer

   >
   {style="tip"}

## Test the app

Now it's time for you to test the app.

> ### Test UI manually

Reviewer: ```suggestion
### Test the UI manually
```

Reviewer: (suggestion) not necessarily needed but to me it reads better with the article

> Start by testing the UI functionality. Complete some simple actions:

Reviewer: ```suggestion
Start by testing the UI functionality. Try some simple actions:
```

Reviewer: (suggestion) just a small suggestion - whichever feels better to you.

![run success icon](app-client.expui.gutter.runSuccess.svg){width=20}.

If any test doesn't work, send a new prompt to Claude asking it to investigate and fix the problem.

## Make refinements

Now that the initial task is complete, you can make refinements. For example, let's improve the UI so users can edit
tasks directly in the list.

You can send a prompt like:

```text
As a next step, allow users to edit tasks inline. For example, let users click on a task title to edit it directly in the list,
and update fields like priority, due date, or status without leaving the current view. 
This change should make the app feel faster and more intuitive to use.
```

Just as before, Claude explores the existing project structure and suggests a plan.
After you accept the plan, wait for Claude to finish, review the changes, and then run the app again.

<img src="make-refinements-claude.gif" alt="Refine your Spring Boot app with Claude" width="600"/>

Congratulations! You used Claude to plan, build, test, and refine a Kotlin Spring Boot application directly in IntelliJ IDEA.

## What's next?

> * Learn about Kotlin AI skills

Reviewer: Just a sanity check - will these be updated later with links, but they are not yet done? 🤔

Author: Yes, exactly. They don't exist yet :)
